src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,623 @@
+ * a separate suppressions file, which must be kept up-to-date as the source
+ * file is edited.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code nearbyTextPattern} - Specify nearby text
+ * pattern to trigger filter to begin suppression.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code "SUPPRESS CHECKSTYLE (\w+)"}.
+ * </li>
+ * <li>
+ * Property {@code checkPattern} - Specify check name pattern to suppress.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code ".*"}.
+ * </li>
+ * <li>
+ * Property {@code messagePattern} - Specify check violation message pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code idPattern} - Specify check ID pattern to suppress.
+ * Type is {@code java.util.regex.Pattern}.
+ * Default value is {@code null}.
+ * </li>
+ * <li>
+ * Property {@code lineRange} - Specify negative/zero/positive value that
+ * defines the number of lines preceding/at/following the suppressing nearby text.
+ * Property can also be a RegExp group index at {@code nearbyTextPattern} in
+ * format of {@code $x} and be picked from target file.
+ * Type is {@code java.lang.String}.
+ * Default value is {@code "0"}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the filter to suppress audit events on the same line:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;/&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;MagicNumber&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * int a = 1; // SUPPRESS CHECKSTYLE the reason for this is..
+ * int b = 1; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to suppress audit events on any line that contains
+ * {@code DO NOT CHECK THIS LINE}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;&gt;
+ *     &lt;property name=&quot;nearbyTextPattern&quot; value=&quot;DO NOT CHECK THIS LINE&quot;/&gt;
+ *   &lt;/module&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;MagicNumber&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * int a = 1; // DO NOT CHECK THIS LINE
+ * int b = 1; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to suppress audit events only on {@code FileTabCharacter}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyTextFilter&quot;&gt;
+ *     &lt;property name=&quot;checkPattern&quot; value=&quot;FileTabCharacter&quot;/&gt;
+ *     &lt;property name=&quot;nearbyTextPattern&quot; value=&quot;.*&quot;/&gt;
```

Comment:
Doesn't this mean all lines are considered a suppression since we don't check if these are in some sort of comment or such?
I think we should explain to the users the dangers of not having a specific text pattern.

Reply:
Done.

Reply:
I think the explanation needs to be more severe. We should flat out say not to do this and our examples should not use it.

Literally any text is seen as a suppression. Your start tag for an XML document will trigger a suppression. It basically amounts to ALL violations (as long as other properties match) being suppressed. If they want to do this, they should suppress the violation in the file itself and that should really be configured with one of the other filters.

Reply:
So remove the example completely?

Reply:
We should not use it as an example of how to use the check. I am fine removing the ones we have completely, but we still need the verbage to warn users about this.

Reply:
I was thinking something like this. Is that okay? Under notes:

![Screenshot from 2023-03-08 09-34-38](https://user-images.githubusercontent.com/23459549/223663239-2626e6de-7fce-4eb1-952a-40a89ff4965c.png)

Reply:
Under notes is ok, yes.

`and will likely suppress all audit events in the file. It is best to set this to a key phrase not commonly used in the file to help denote it out of the rest of the file as a suppression. See the default value as an example.`.

Reply:
Done.